src/scrcmd_message.c

```diff
@@ -0,0 +1,310 @@
+}
+
+void ovFieldMain_ShowMessageInField(SCRIPTCONTEXT *ctx, MSGDATA *messageData, u32 messageNum) {
+    MessageBox messageBox;
+    ovFieldMain_GetMsgBoxParameters(ctx->fsys, &messageBox);
+    ovFieldMain_CreateMessageBox(ctx->fsys, &messageBox);
+    ovFieldMain_ReadAndExpandMsgDataViaBuffer(&messageBox, messageData, messageNum);
+    ovFieldMain_AddTextPrinterParameterized(&messageBox, 1);
+}
+
+// word2 needs to be signed to match
+void ov01_021EF564(SCRIPTCONTEXT *ctx, u16 messageBank, u16 messageNum, u16 word1, s16 word2, u8 canABSpeedUp) {
+    MessageBox messageBox;
+    ovFieldMain_GetMsgBoxParameters(ctx->fsys, &messageBox);
+    ovFieldMain_CreateMessageBox(ctx->fsys, &messageBox);
+    ovFieldMain_GetFormattedECMessage(&messageBox, messageBank, messageNum, word1, (u16)word2);
+    if (canABSpeedUp != 0xFF) {
+        ov01_021EF758(&messageBox, 1, ovFieldMain_GetTextFrameDelay(ctx), canABSpeedUp, 0);
+    } else {
+        ovFieldMain_AddTextPrinterParameterized(&messageBox, 1);
+    }
+}
+
+void ov01_021EF5C8(SCRIPTCONTEXT *ctx, MessageFormat *messageFormat, u8 messageNum, u32 canABSpeedUp) {
+    MessageBox messageBox;
+    ovFieldMain_GetMsgBoxParametersEx(ctx->fsys, messageFormat, &messageBox);
+    ovFieldMain_CreateMessageBox(ctx->fsys, &messageBox);
+    ovFieldMain_ReadAndExpandMsgDataViaBuffer(&messageBox, ctx->msgdata, messageNum);
+    ov01_021EF758(&messageBox, 1, ovFieldMain_GetTextFrameDelay(ctx), canABSpeedUp, 0);
+}
+
+u32 ovFieldMain_GetTextFrameDelay(SCRIPTCONTEXT *ctx) {
+    return Options_GetTextFrameDelay(Sav2_PlayerData_GetOptionsAddr(ctx->fsys->savedata));
+}
+
+void ovFieldMain_GetMsgBoxParameters(FieldSystem *fsys, MessageBox *messageBox) {
+    messageBox->message = *(STRING **)FieldSysGetAttrAddr(fsys, SCRIPTENV_STRING_BUFFER_0);
+    messageBox->buffer = *(STRING **)FieldSysGetAttrAddr(fsys, SCRIPTENV_STRING_BUFFER_1);
+    messageBox->messageFormat = *(MessageFormat **)FieldSysGetAttrAddr(fsys, SCRIPTENV_MESSAGE_FORMAT);
+    messageBox->window = (WINDOW *)FieldSysGetAttrAddr(fsys, SCRIPTENV_WINDOW);
+    messageBox->unk10 = (u8 *)FieldSysGetAttrAddr(fsys, SCRIPTENV_FIELD_08);
+    messageBox->textPrinterNumPtr = (u8 *)FieldSysGetAttrAddr(fsys, SCRIPTENV_TEXT_PRINTER_NUMBER);
+}
+
+void ovFieldMain_GetMsgBoxParametersEx(FieldSystem *fsys, MessageFormat *messageFormat, MessageBox *messageBox) {
+    messageBox->message = *(STRING **)FieldSysGetAttrAddr(fsys, SCRIPTENV_STRING_BUFFER_0);
+    messageBox->buffer = *(STRING **)FieldSysGetAttrAddr(fsys, SCRIPTENV_STRING_BUFFER_1);
+    messageBox->messageFormat = messageFormat;
+    messageBox->window = (WINDOW *)FieldSysGetAttrAddr(fsys, SCRIPTENV_WINDOW);
+    messageBox->unk10 = (u8 *)FieldSysGetAttrAddr(fsys, SCRIPTENV_FIELD_08);
+    messageBox->textPrinterNumPtr = (u8 *)FieldSysGetAttrAddr(fsys, SCRIPTENV_TEXT_PRINTER_NUMBER);
+}
+
+void ovFieldMain_CreateMessageBox(FieldSystem *fsys, MessageBox *messageBox) {
+    if (*messageBox->unk10 == 0) {
```

Comment:
please use brackets to make it clear what is being dereferenced (and on similar ones)